client/src/components/profile/components/HeatMap.js

```diff
@@ -22,7 +22,22 @@ const propTypes = {
 };
 
 function HeatMap({ calendar, streak }) {
-  const startOfToday = startOfDay(Date.now());
+  // an issue with react-calendar-heatmap makes the days off by one
+  // see this https://github.com/kevinsqi/react-calendar-heatmap/issues/112
+  // I have added one day in the marked places to account for the offset
+
+  // this logic adds a day to all the timestamps (remove if issue gets fixed)
+  let tempCalendar = {};
+  const secondsInADay = 60 * 60 * 24;
+  for (let timestamp in calendar) {
```

Comment:
Maybe use `forEach` as `for...in` [isn't the best for this sort of thing](https://stackoverflow.com/a/3010848).

If you are going to change this, probably best to use `forEach`  here https://github.com/freeCodeCamp/freeCodeCamp/blob/c86903a0e121a543627a071658667ee5991bed80/client/src/components/profile/components/HeatMap.js#L53
as well.  It should all end up a little more concise, as you don't need to check if the ids exist and so on.

Reply:
Thanks for looking at this, `calendar` is simply an object and not an array that looks like this...
```
{
  1567696744262: 1,
  1567696744264: 1,
  ...
}
```
So I can't use the array methods. I would have to grab the keys or something - I can take a look and see if I can find a better way.

Reply:
Oh, I assumed it was an array - my mistake.  Weird that it's done this way, but it is what it is.

In that case, is there any need for the following?
https://github.com/freeCodeCamp/freeCodeCamp/blob/c86903a0e121a543627a071658667ee5991bed80/client/src/components/profile/components/HeatMap.js#L54

Anyway, sorry for the misunderstanding.

Reply:
I agree that it does look a little sloppy and could probably be better somehow. I was thinking about taking a deeper look at where that data is coming from and maybe trying to make things a little more efficient or something. It just feels a little weird and like it could be better.

> In that case, is there any need for the following?

I didn't think so, but eslint was having a fit when I didn't put it in...
<img width="586" alt="Screen Shot 2020-01-30 at 5 33 30 PM" src="https://user-images.githubusercontent.com/20648924/73499786-a9b34c00-4386-11ea-82c5-dabb5de50a01.png">

Reply:
`calendar` comes from here:
https://github.com/freeCodeCamp/freeCodeCamp/blob/2f39dd07d66c55d498399a6fc6cfe21e196ebd2f/api-server/server/utils/publicUserProps.js#L70-L75
so it's just a plain object of the form
```js
{
  "<time-in-seconds>": 1,
  ...
}
```

How about:
```js
  for (let timestamp of Object.keys(calendar)) {
```
since we only want keys and `Object.keys` ignores everything that's not directly on the object.

Reply:
Well, I tried to simplify some of the logic and cut out a step. Seems a little better.

I don't think a simple snapshot will work here because the days of the heatmap will always change. Meaning the snapshot taken today will show 2/1 as the most recent day on the calendar/snapshot. But tomorrow the snapshot won't match anymore.